src/ng/rootScope.js

```diff
@@ -786,12 +786,12 @@ function $RootScopeProvider() {
           current = target;
 
           // It's safe for asyncQueuePosition to be a local variable here because this loop can't
-          // be reentered recursively. Calling $digest from a function passed to $applyAsync would
+          // be reentered recursively. Calling $digest from a function passed to $evalAsync would
           // lead to a '$digest already in progress' error.
           for (var asyncQueuePosition = 0; asyncQueuePosition < asyncQueue.length; asyncQueuePosition++) {
             try {
               asyncTask = asyncQueue[asyncQueuePosition];
-              asyncTask.scope.$eval(asyncTask.expression, asyncTask.locals);
+              asyncTask.expression(asyncTask.scope, asyncTask.locals);
```

Comment:
If you mean string expressions like `this.foo`, no, it won't affect them:
```js
angular.injector(['ng']).get('$parse')('this.foo').call({foo: 1}, {foo: 2}); // 2
```
If you mean functions passed as expressions to `$evalAsync`, it will affect them, but does this matter?